src/averageAges.js

```diff
@@ -37,7 +42,16 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = withChildren
+    ? people.filter((p) => p.sex === 'f'
+      && people.find((p1) => p1.mother === p.name))
+    : people.filter((p) => p.sex === 'f');
```

Comment:
```suggestion
  const filteredWomen =  people.filter((p) => p.sex === 'f')
  const women = withChildren
    ? filteredWomen.filter(({ name }) => people.some(({ mother }) => name === mother))
    : filteredWomen;
```

Reply:
ideally you will create additional function to filter people list by sex and reuse it in `calculateMenAverageAge` as well